packages/foam-core/src/utils.ts

```diff
@@ -1,4 +1,27 @@
+import path from 'path';
+import crypto from 'crypto';
 import { titleCase } from 'title-case';
+import GithubSlugger from 'github-slugger';
+import { URI, ID } from 'types';
+
+export const hash = (text: string) => crypto.createHash('md5').update(text).digest("hex")
+
+export const uriToSlug = (noteUri: URI): string => {
+  // TODO hack don't merge with this
+  const filename = noteUri.split('/').slice(-1)[0].split('.').slice(0, -1).join('.');
+  return GithubSlugger.slug(filename);
+}
+
+export const hashURI = (uri: URI): ID => {
+  return hash(path.normalize(uri));
+}
+
+export const getUriViaRelative = (reference: URI, relativeSlug: string): URI => {
+  const slug = relativeSlug.endsWith('.md') // TODO hack
```

Comment:
this is a hack, I am assuming an `md` extension, which of course we can't.
probably need to add a proper extension check, will do that while you take a look at the approach as a whole